src/tablib/formats/_ods.py

```diff
@@ -53,45 +53,18 @@ def dset_sheet(cls, dataset, ws):
             _offset = i
             _package.insert((sep[0] + _offset), (sep[1],))
 
-        for i, row in enumerate(_package):
-            row_number = i + 1
-            odf_row = table.TableRow(stylename=bold, defaultcellstylename='bold')
+        for row_number, row in enumerate(_package, start=1):
+            is_header = row_number == 1 and dataset.headers
+            style = bold if is_header else None
+            odf_row = table.TableRow(stylename=style)
+            ws.addElement(odf_row)
             for j, col in enumerate(row):
-                try:
-                    col = str(col, errors='ignore')
-                except TypeError:
-                    # col is already str
-                    pass
-                ws.addElement(table.TableColumn())
-
-                # bold headers
-                if (row_number == 1) and dataset.headers:
-                    odf_row.setAttribute('stylename', bold)
-                    ws.addElement(odf_row)
-                    cell = table.TableCell()
-                    p = text.P()
-                    p.addElement(text.Span(text=col, stylename=bold))
-                    cell.addElement(p)
-                    odf_row.addElement(cell)
-
-                # wrap the rest
+                if isinstance(col, (int, float)):
```

Comment:
In my original usecase, I was using `decimal.Decimal` values, which are not handled by your code now. Changing this line as shown below should catch all numeric values (as [suggested here](https://stackoverflow.com/a/4187266/740048)) and makes your PR work perfectly for my case with `Decimal` values (of course the import should go at the top).

```suggestion
                import numbers
                if isinstance(col, numbers.Number):
```